### PR TITLE
Refactor to Azure Workspace Module

### PR DESCRIPTION
**Pull request by shreejam21**
- [x] Refactored the code to just have Workspace Logic
- [x] Refactored Examples
- [x] Rewrote READMEs
- [x] Tested Examples

### CODE REVIEW

File: examples/databricks-workspace-cluster/README.md [hunk 1]
```diff
@@ -18,27 +18,45 @@ The module can deploy an Intel Optimized Azure Databricks Workspace and Cluster.
...
+
+  2.  See the main.tf in the example on how to pass the value for dbx_host (i.e. the URL of the databricks workspace) in the Databricks Cluster Module 
+
+  3.  Example of main.tf
+
+        ```hcl
+        # Provision Intel Cloud Optimization Module for Azure Databricks
+
+        module "module-example" {
+            source                  = "intel/azure-databricks/intel"
```
Comment (lucasmelogithub) [inline examples/databricks-workspace-cluster/README.md:37]: update to azure-databricks-workspace

File: examples/databricks-workspace-cluster/README.md [hunk 2]
```diff
@@ -18,27 +18,45 @@ The module can deploy an Intel Optimized Azure Databricks Workspace and Cluster.
...
+            resource_group_name     = <"ENTER_YOUR_RESOURCE_GROUP_NAME">                #Required. 
+            dbx_workspace_name      = <"NAME_YOUR_DATABRICKS_WORKSPACE">                #Required. 
+        }
+
+        # This example creates databricks cluster on your azure dbx workspace created above.
+
+        module "databricks_cluster" {
+            source     = "../../"
```
Comment (lucasmelogithub) [inline examples/databricks-workspace-cluster/README.md:45]: update to source = "intel/databricks-cluster/intel"

File: examples/databricks-workspace-custom-vnet/main.tf [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ## This module deploys a Azure Databricks workspace with an Intel Optimized Cluster in a user provided resource group and virtual network
-module "azure-dbx-workspace" {
-  source              = "intel/azure-databricks/intel"
+module "azure_databricks_workspace" {
+  source              = "../../"
```
Comment (lucasmelogithub) [inline examples/databricks-workspace-custom-vnet/main.tf:3]: update source = "intel/azure-databricks-workspace/intel"

**Comment by lucasmelogithub** [comment]
Please see comments @shreejam21

**Comment by shreejam21** [comment]
made the changes base don comment

**Review by lucasmelogithub — APPROVED** [review]
LGTM, approving.

**Pull request merged**